### PR TITLE
[CASSH SERVER][1.6.1] Add exception for admin check and fix principals

### PR DESCRIPTION
**Pull request by nbeguier**
- Add exception for admin check
  - fix principals

### CODE REVIEW

File: src/server/server.py [hunk 1]
```diff
@@ -228,11 +228,15 @@ def ldap_authentification(admin=False):
...
         except Exception as e:
             return False, 'Error: %s' % e
-        if admin and SERVER_OPTS['ldap_admin_cn'] not in\
-            ldap_conn.search_s(SERVER_OPTS['ldap_bind_dn'], 2,
-                               filterstr='(%s=%s)' % (SERVER_OPTS['filterstr'], realname)
-                              )[0][1]['memberOf']:
-            return False, 'Error: user %s is not an admin.' % realname
+        if admin:
+            try:
+                memberof_list = ldap_conn.search_s(SERVER_OPTS['ldap_bind_dn'], 2,
```
Comment (aladjadj) [inline src/server/server.py]: replace '2' by a CONSTANT ( ex: ldap.SCOPE_SUBTREE )

File: src/server/server.py [hunk 2]
```diff
@@ -228,11 +228,15 @@ def ldap_authentification(admin=False):
...
+                                                   filterstr='(%s=%s)' % (SERVER_OPTS['filterstr'], realname)
```
Comment (aladjadj) [inline src/server/server.py]: maybe replace by filter
'(&(%s=%s)(memberOf=%s))'  % (SERVER_OPTS['filterstr'], realname, SERVER_OPTS['ldap_admin_cn'])
and except ldap.NO_SUCH_OBJECT

**Review by aladjadj — CHANGES_REQUESTED** [review]

File: src/server/server.py (hunk 1 above)
Comment (nbeguier) [reply src/server/server.py]: great !

File: src/server/server.py (hunk 2 above)
Comment (nbeguier) [reply src/server/server.py]: Good point.
But, the return object is always a list of users, so there is no need of exception.

**Comment by nbeguier** [comment]
@aladjadj Files are updated, thanks for your help

File: src/server/server.py [hunk 3]
```diff
@@ -228,11 +228,16 @@ def ldap_authentification(admin=False):
...
+        if admin:
+            memberof_admin_list = ldap_conn.search_s(
+                SERVER_OPTS['ldap_bind_dn'],
+                SCOPE_SUBTREE,
+                filterstr='(&(%s=%s)(memberOf=%s))' % (
+                    SERVER_OPTS['filterstr'],
+                    realname,
+                    SERVER_OPTS['ldap_admin_cn']))
+            if not memberof_admin_list:
+                return False, 'Error: user %s is not an admin.' % realname
```
Comment (xakraz) [inline src/server/server.py:240]: :+1:

**Review by xakraz — APPROVED** [review]

File: src/server/server.py (hunk 3 above)
Comment (aladjadj) [reply src/server/server.py:240]: :+1:

**Review by aladjadj — APPROVED** [review]

**Pull request merged**